lib/reig/context.cpp

```diff
@@ -97,85 +97,52 @@ namespace reig {
-            } else {
-                _previous_windows.insert(_previous_windows.begin(), queued_window);
-            }
-        }
-
+    void Context::update_window_layers() {
         if (!mouse.left_button.is_clicked()) return;
 
-        for (auto it = _previous_windows.begin(); it != _previous_windows.end(); ++it) {
+        for (auto it = _windows.begin(); it != _windows.end(); ++it) {
             if (mouse.left_button.just_clicked_in_window(detail::get_full_window_rect(*it))) {
-                std::iter_swap(it, _previous_windows.begin());
+                std::iter_swap(it, _windows.begin());
                 break;
             }
         }
     }
 
-    void Context::cleanup_previous_windows() {
-        auto remove_from = std::remove_if(_previous_windows.begin(), _previous_windows.end(),
-                                         [this](const Window& previous_window) {
-                                             return std::none_of(_queued_windows.begin(), _queued_windows.end(),
-                                                                     [&previous_window](const Window& window) {
-                                                                         return window.id == previous_window.id;
-                                                                     });
-                                         });
-        _previous_windows.erase(remove_from, _previous_windows.end());
+    void Context::remove_unqueued_windows() {
+        auto remove_from = std::remove_if(_windows.begin(), _windows.end(), [](const Window& window) {
+            return !window.is_queued();
+        });
+        _windows.erase(remove_from, _windows.end());
     }
 
     bool Context::handle_window_focus(const Window& window, bool is_claiming) {
         if (is_claiming) {
             if (!_dragged_window) {
-                _dragged_window = window.id;
+                _dragged_window = window.id();
             }
-            return _dragged_window == window.id;
+            return _dragged_window == window.id();
         } else {
-            if (_dragged_window == window.id) {
+            if (_dragged_window == window.id()) {
                 _dragged_window = nullptr;
             }
-            return _dragged_window != window.id;
+            return _dragged_window != window.id();
         }
     }
 
-    void Context::start_frame() {
-        mouse.left_button._is_clicked = false;
-        mouse._scrolled = 0.f;
-
-        keyboard.reset();
-
-        ++_frame_counter;
-    }
-
-    unsigned Context::get_frame_counter() const {
-        return _frame_counter;
-    }
-
     detail::Window* Context::get_current_window() {
-        if (!_queued_windows.empty() && !_queued_windows.back().is_finished) {
-            return &_queued_windows.back();
+        auto current_window = std::find_if(_windows.begin(), _windows.end(), [](const Window& window) {
+            return window.is_queued() && !window.is_finished();
+        });
```

Comment:
A pointer to current window can be stored, that is updated on each start_window and end_window, so that a search is not needed